page-factory-core/src/main/java/ru/sbtqa/tag/pagefactory/data/DataParser.java

```diff
@@ -59,15 +39,23 @@ public void replaceStepPlaceholders(List<CucumberFeature> cucumberFeatures) thro
 
                 for (Step step : steps) {
                     FieldUtils.writeField(step, "argument", replaceArgumentPlaceholders(step.getArgument()), true);
-                    FieldUtils.writeField(step, "text", replaceStepPlaceholders(step.getText()), true);
+                    FieldUtils.writeField(step, "text", replaceDataPlaceholders(step.getText()), true);
                 }
             }
         }
     }
 
+    private List<Tag> getScenarioTags(ScenarioDefinition scenarioDefinition) {
+        try {
+            return (List<Tag>) FieldUtils.readField(scenarioDefinition, "tags", true);
+        } catch (IllegalArgumentException | IllegalAccessException e) {
+            return new ArrayList<>();
```

Comment:
R we gonna mutate this one? If no I suggest to use Collections.EMPTY_LIST to avoid memory allocation